src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSerializationVisitor.cs

```diff
@@ -502,18 +527,49 @@ protected override void ProcessNullConstantExpression(IEdmNullExpression express
         protected override void ProcessCastExpression(IEdmCastExpression expression)
         {
             bool inlineType = IsInlineType(expression.Type);
-            this.BeginElement(expression, (IEdmCastExpression t) => { this.schemaWriter.WriteCastExpressionElementHeader(t, inlineType); }, e => { this.ProcessFacets(e.Type, inlineType); });
-            if (!inlineType)
+
+            bool isXml = this.schemaWriter is EdmModelCsdlSchemaXmlWriter;

```

Comment:
Perhaps you should set isXml once, as a private property, so you don't have to repeatedly call "is"?